File: src/main/java/cz/grunwjir/csasdemo/csas/client/CsasClientProperties.java
package cz.grunwjir.csasdemo.csas.client;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * Configuration properties class for Csas API client.
 *
 * @author Jiri Grunwald
 */
@Getter
@Setter
@ConfigurationProperties(prefix = "csas")
public class CsasClientProperties {

    private String host;
    private String apiKey;
}

File: src/main/kotlin/cz/grunwjir/csasdemo/csas/client/CsasClientProperties.kt
package cz.grunwjir.csasdemo.csas.client

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

/**
 * Configuration properties class for Csas API client.
 *
 * @author Jiri Grunwald
 */
@ConfigurationProperties(prefix = "csas")
@ConstructorBinding
data class CsasClientProperties(val host: String, val apiKey: String)